src/platform/cmake/FindLuaJIT.cmake

```diff
@@ -0,0 +1,319 @@
+#[=======================================================================[.rst:
+FindLuaJIT
+-------
+
+Locate LuaJIT library.
+
+This module defines:
+
+``LUAJIT_FOUND``
+  if false, do not try to link to LuaJIT
+``LUAJIT_LIBRARIES``
+  both luajit and luajitlib
+``LUAJIT_INCLUDE_DIR``
+  where to find lua.h
+``LUAJIT_VERSION_STRING``
+  the version of LuaJIT found
+``LUAJIT_VERSION_MAJOR``
+  the major version of LuaJIT
+``LUAJIT_VERSION_MINOR``
+  the minor version of LuaJIT
+``LUAJIT_VERSION_PATCH``
+  the patch version of LuaJIT
+
+Note that the expected include convention is
+
+::
+
+  #include "lua.h"
+
+and not
+
+::
+
+  #include <lua/lua.h>
+
+This is because, the lua location is not standardized and may exist in
+locations other than lua/
+#]=======================================================================]
+
+cmake_policy(PUSH)  # Policies apply to functions at definition-time
+cmake_policy(SET CMP0012 NEW)  # For while(TRUE)
+
+unset(_luajit_include_subdirs)
+unset(_luajit_library_names)
+unset(_luajit_append_versions)
+
+# this is a function only to have all the variables inside go away automatically
+function(_lua_get_versions)
```

Comment:
This is irrelevant for LuaJIT as it always acts like 5.1.

Reply:
I believe that's not true for LuaJIT 2.1, but also this file has a lot of stuff that definitely mishandles the versioning as it expects Lua versions

Reply:
LuaJIT 2.1 still always acts like 5.1 in terms of semantics; it just supports some 5.2+ features as extensions.

Reply:
That's because inside LuaJIT are defined LUAJIT_VERSION (2.x) and LUA_VERSION (5.1). LUA_VERSION is used into the library name. I've simplified the cmake file. As previously said LuaJIT'll always be Lua 5.1, all non LuaJIT references will be removed once I commit hte new changes.